model/shared/src/main/scala/explore/model/TimingWindow.scala

```diff
@@ -106,6 +110,48 @@ case class TimingWindow private (
 
   def repeatForever: Boolean =
     repetition.exists(_.toOption.exists(_.repeatPeriod.exists(_.repeatFrequency.isEmpty)))
+
+  def toIntervalSeq(within: BoundedInterval[Instant]): IntervalSeq[Instant] = {
```

Comment:
I wonder if this expansion logic should go to core, we probably will need it for the scheduler

Reply:
Do we want the model classes in core? In that case the method will go with them.